### ISSUE TITLE
Component "name" in className with `.extend`

### ISSUE THREAD

**Issue by breadadams** [styled-components/babel-plugin-styled-components#145]
It seems that with an extended component, the component being extended's name/filename (I've tried with both config options) are used for the classname applied to the DOM element. Ie:

```jsx
const Example = styled.div`
  color: red;
`
const ExampleExtended = Example.extend`
  background: green;
`
...

render() {
  return <ExampleExtended />
}
```

The above returns something like:
```html
<div class="Example-cJRMOC jmwXEm"></div>
```

Whereas ideally I was expecting:
```html
<div class="ExampleExtended-cJRMOC jmwXEm"></div>
```

Is this intended functionality?

___

##### Versions
```
"styled-components": "^3.3.3",
"babel-plugin-styled-components": "^1.5.1",
```

**Comment by breadadams** [comment]
Here's a _real-world example_ (we're using a base "building block" to construct other components via `.extend`):

#### Before (using `const Component = styled.div...`)
![image](https://user-images.githubusercontent.com/5795227/41997753-30521a32-7a59-11e8-8f05-b501108fcf72.png)

#### After (using `.extend`)
![image](https://user-images.githubusercontent.com/5795227/41997856-7a9cc4e8-7a59-11e8-8e37-ef76a1d6b51e.png)

As you can see we lose the debugging benefit, as all component names are replaced with the component used in the `.extend` (even the "parent component" names are gone 🤷‍♂️ )

**Comment by probablyup** [comment]
Hmm this is probably a bug.

**Comment by breadadams** [comment]
Happy to take a look however it may be past my abilities... It seems `/src/visitors/displayNameAndId.js` is taking care of the naming @probablyup?

**Comment by probablyup** [comment]
Yeah that should be correct. You’ll also want to upgrade whatever version
of s-c we’re testing with locally since there were some recent changes to
displayName handling.

**Comment by breadadams** [comment]
I've done some testing on the issue and don't appear to be getting very far, however I'll share my findings incase it can assist someone else in coming up with a solution 🙂 

I updated [the tests](https://github.com/styled-components/babel-plugin-styled-components/blob/master/test/fixtures/17-annotate-css-with-uglify-pure-comments/index.js#L59) to look like so:

![image](https://user-images.githubusercontent.com/5795227/42138566-95b7a4f4-7d7f-11e8-9716-529ee2ce01bc.png)

The [`path` object](https://github.com/styled-components/babel-plugin-styled-components/blob/master/src/utils/getName.js#L14) (received in `getName`) for an extended component only includes the _real_ name (not the name of the component from which it was extended) in 3 places, `references`, `bindings` & `code`:

![image](https://user-images.githubusercontent.com/5795227/42138577-cd473bd2-7d7f-11e8-8c5b-9cb45a97902f.png)
![image](https://user-images.githubusercontent.com/5795227/42138578-d561d05c-7d7f-11e8-8731-9a00c80a45d3.png)

And therefore, the actual `name` assigned to `DoubleInterpolation` is `TestToBeExtended`:

![image](https://user-images.githubusercontent.com/5795227/42138590-19fb9752-7d80-11e8-915f-286433aa7a54.png)

Could it be that the source of this bug doesn't lie within the babel plugin but SC itself? 🤔

**Comment by breadadams** [comment]
Also @probablyup I couldn't detect which version of styled-components was being used locally, it doesn't appear to be in `package.json`, _or_ my `yarn.lock` after init. Something I'm missing?

**Comment by probablyup** [comment]
@breadadams `yarn list styled-components` should give you the answer

As a general FYI, the "extend" API is deprecated and will be removed in v4, so I'd recommend switching to `styled(StyledComponent)` instead which should not have any issues with classnames. https://github.com/styled-components/styled-components/issues/1694
